### ISSUE TITLE
Unknown error with kinesis

### ISSUE THREAD

**Issue by cetanu** [timberio/vector#1146]
Getting the following message repeating:

```
Nov 05 06:18:41 ip-10-108-1-171 vector[30282]: Nov 05 06:18:41.117 ERROR sink{name="json_to_kinesis"}: vector::sinks::util::retries: encountered non-retriable error. error=unknown error
Nov 05 06:18:41 ip-10-108-1-171 vector[30282]: Nov 05 06:18:41.117 ERROR sink{name="json_to_kinesis"}: vector::sinks::util: request failed. error=unknown error
```

as far as I can tell, nothing is actually reaching kinesis

Debug logs show the following happens prior to the error:

```
Nov 05 06:19:52.265 DEBUG sink{name="json_to_kinesis"}:request: tokio_reactor::registration: scheduling Read for: 0
Nov 05 06:19:52.266 DEBUG sink{name="json_to_kinesis"}:request: tokio_reactor: dropping I/O source: 0
Nov 05 06:19:52.266 DEBUG sink{name="json_to_kinesis"}:request: tokio_reactor: dropping I/O source: 1
Nov 05 06:19:52.266 DEBUG sink{name="json_to_kinesis"}:request: tokio_reactor: dropping I/O source: 2
Nov 05 06:19:52.266 DEBUG sink{name="json_to_kinesis"}:request: rusoto_core::signature: Creating request to send to AWS.
Nov 05 06:19:52.266 DEBUG sink{name="json_to_kinesis"}:request: rusoto_core::request: Full request:
 method: POST
 final_uri: https://kinesis.eu-west-1.amazonaws.com/
 payload: {"Records":[{"Data":"REDACTED=","PartitionKey":"񥢔򃿉𙄡􁔍𣼊򼥆򫯱򶲂󵗷󁔔񆶔𸆵񛄾𤁙񎿭󤜭"}
,"StreamName":"prod-logs"}
Headers:

Nov 05 06:19:52.266 DEBUG sink{name="json_to_kinesis"}:request: rusoto_core::request: authorization:"AWS4-HMAC-SHA256 Credential=REDACTED/eu-west-1/kinesis/aws4_request, SignedHeaders=content-type;host;x-amz-date;x
-amz-target, Signature=REDACTED"
Nov 05 06:19:52.266 DEBUG sink{name="json_to_kinesis"}:request: rusoto_core::request: content-length:"33191"
Nov 05 06:19:52.266 DEBUG sink{name="json_to_kinesis"}:request: rusoto_core::request: content-type:"application/x-amz-json-1.1"
Nov 05 06:19:52.266 DEBUG sink{name="json_to_kinesis"}:request: rusoto_core::request: host:"kinesis.eu-west-1.amazonaws.com"
Nov 05 06:19:52.266 DEBUG sink{name="json_to_kinesis"}:request: rusoto_core::request: x-amz-content-sha256:"REDACTED"
Nov 05 06:19:52.266 DEBUG sink{name="json_to_kinesis"}:request: rusoto_core::request: x-amz-date:"20191105T061952Z"
Nov 05 06:19:52.266 DEBUG sink{name="json_to_kinesis"}:request: rusoto_core::request: x-amz-target:"Kinesis_20131202.PutRecords"
Nov 05 06:19:52.266 DEBUG sink{name="json_to_kinesis"}:request: rusoto_core::request: user-agent:"rusoto/0.37.0 rust/1.38.0 linux"
Nov 05 06:19:52.267 DEBUG hyper::client::connect::dns: resolving host="kinesis.eu-west-1.amazonaws.com"
Nov 05 06:19:52.267 DEBUG sink{name="json_to_kinesis"}:request: hyper::client::connect::http: connecting to 52.94.220.26:443
Nov 05 06:19:52.267 DEBUG sink{name="json_to_kinesis"}:request: tokio_reactor: adding I/O source: 12582914
Nov 05 06:19:52.267 DEBUG sink{name="json_to_kinesis"}:request: tokio_reactor::registration: scheduling Write for: 2
Nov 05 06:19:52.294 DEBUG sink{name="json_to_kinesis"}:request: hyper::client::connect::http: connected to Some(V4(52.94.220.26:443))
Nov 05 06:19:52.294 DEBUG sink{name="json_to_kinesis"}:request: tokio_reactor::registration: scheduling Read for: 2
Nov 05 06:19:52.294 DEBUG sink{name="json_to_kinesis"}:request: tokio_reactor::registration: scheduling Read for: 2
Nov 05 06:19:52.320 DEBUG sink{name="json_to_kinesis"}:request: tokio_reactor::registration: scheduling Read for: 2
Nov 05 06:19:52.321 DEBUG sink{name="json_to_kinesis"}:request: tokio_reactor::registration: scheduling Read for: 2
Nov 05 06:19:52.321 DEBUG sink{name="json_to_kinesis"}:request: tokio_reactor::registration: scheduling Read for: 2
Nov 05 06:19:52.347 DEBUG tokio_reactor::registration: scheduling Read for: 2
Nov 05 06:19:52.348 DEBUG tokio_reactor::registration: scheduling Read for: 2
Nov 05 06:19:52.348 DEBUG hyper::proto::h1::io: flushed 567 bytes
Nov 05 06:19:52.348 DEBUG hyper::proto::h1::io: flushed 16384 bytes
Nov 05 06:19:52.348 DEBUG hyper::proto::h1::io: flushed 16384 bytes
Nov 05 06:19:52.348 DEBUG hyper::proto::h1::io: flushed 423 bytes
Nov 05 06:19:52.404 DEBUG hyper::proto::h1::io: read 312 bytes
Nov 05 06:19:52.404 DEBUG hyper::proto::h1::io: parsed 6 headers
Nov 05 06:19:52.404 DEBUG hyper::proto::h1::conn: incoming body is content-length (107 bytes)
Nov 05 06:19:52.404 DEBUG hyper::proto::h1::io: read 107 bytes
Nov 05 06:19:52.404 DEBUG hyper::proto::h1::conn: incoming body completed
Nov 05 06:19:52.404 DEBUG tokio_reactor: dropping I/O source: 2
Nov 05 06:19:52.404 ERROR sink{name="json_to_kinesis"}: vector::sinks::util::retries: encountered non-retriable error. error=unknown error
Nov 05 06:19:52.404 ERROR sink{name="json_to_kinesis"}: vector::sinks::util: request failed. error=unknown error
```

This is the configuration of the sink:

```
[sinks.json_to_kinesis]
  type = "aws_kinesis_streams"
  inputs = ["example_source"]
  region = "eu-west-1"
  stream_name = "prod-logs"
  encoding = "text"  # Also tried with json, which is what the logs are
  healthcheck = false  # Because I don't have access to ListStreams
```

Any ideas what this mysterious error message is caused by?

It goes away when I manually specify a partition key, but then it gives messages like "the partition key does not exist" etc...

**Comment by lukesteensen** [comment]
Sorry about this @cetanu! I believe we recently updated a dependency specifically to improve this `unknown error` case. @LucioFranco may be able to confirm, but if you're able to try a recent nightly version of vector hopefully you can get a more helpful error message here.

**Comment by LucioFranco** [comment]
Yeah, with a newer version of `vector` you should actually see the body output. This could be due to a difference between localstack and real kinesis.

**Comment by cetanu** [comment]
Ah, this is completely fixed in 0.6.0. No further troubleshooting required (aka. no errors to troubleshoot).

**Comment by lukesteensen** [comment]
Sounds good! I'll go ahead and close this since you're not seeing any further issues.
